File: src/Draft_Implementation/utils/RunFilter.py
from filter import ESKF, Data
import pandas as pd

data = Data()
TrueFilter = ESKF(data)

def run_filter():
    while TrueFilter.iteration < 5000:
        TrueFilter.predict()
        TrueFilter.update()

    #TrueFilter.save_results('quaternion_results.csv') ~~ csv loading only

File: src/Final_Implementation/utils/RunFilter.cpp
//entry point for running the filter

#include "eskf/utils/RunFilter.h"
#include "eskf/utils/Data.h"
#include "eskf/filter/filter.h"

void RunFilter() {
    const data = Data();
    ESKF TrueFilter = new ESKF(data);
    while(TrueFilter.itteration < 5000) {
        TrueFilter.predict();
        TrueFilter.update();
    }
}

